packages/walletconnect/src/index.ts

```diff
@@ -73,81 +55,81 @@ export class WalletConnect extends Connector {
     if (error) this.onError?.(error)
   }
 
-  private chainChangedListener = (chainId: number | string): void => {
-    this.actions.update({ chainId: parseChainId(chainId) })
+  private chainChangedListener = (chainId: number): void => {
+    this.actions.update({ chainId })
   }
 
   private accountsChangedListener = (accounts: string[]): void => {
     this.actions.update({ accounts })
   }
 
-  private URIListener = (_: Error | null, payload: { params: string[] }): void => {
-    this.events.emit(URI_AVAILABLE, payload.params[0])
+  private URIListener = (uri: string): void => {
+    this.events.emit(URI_AVAILABLE, uri)
   }
 
-  private async isomorphicInitialize(chainId = this.defaultChainId): Promise<void> {
-    if (this.eagerConnection) return
+  private async isomorphicInitialize(): Promise<MockWalletConnectProvider> {
+    if (this.eagerConnection) return this.eagerConnection
 
-    // because we can only use 1 url per chainId, we need to decide between multiple, where necessary
-    const rpc = Promise.all(
-      Object.keys(this.rpc).map(
-        async (chainId): Promise<[number, string]> => [
-          Number(chainId),
-          await getBestUrl(this.rpc[Number(chainId)], this.timeout),
-        ]
+    if (this.provider) {
+      await this.deactivate()
+    }
+    
+    const rpcMap = this.rpcMap
+    const resolvedRpcMap = rpcMap ? Object.fromEntries(
+      await Promise.all(
+        Object.entries(rpcMap).map(
+          async ([chainId, map]): Promise<[string, string]> => [
+            `${chainId}`,
+            await getBestUrl(map, this.timeout),
+          ]
+        )
       )
-    ).then((results) =>
-      results.reduce<{ [chainId: number]: string }>((accumulator, [chainId, url]) => {
-        accumulator[chainId] = url
-        return accumulator
-      }, {})
-    )
-
+    ) : undefined
+    
     return (this.eagerConnection = import('@walletconnect/ethereum-provider').then(async (m) => {
-      this.provider = new m.default({
+      const provider = this.provider = await m.default.init({
```

Comment:
is this change just so you don't have to write `this.provider` below?

Reply:
Haha, yes. But also, I wanted to create a local reference `cost provider` instead of `this.provider` so that I can satisfy Typescript and get rid of `?` property access. With that code, TypeScript is happy (and we are too) that `provider` is always defined.

Reply:
ahhh makes sense 👍